sdk/identity/azure-identity/azure/identity/_credentials/azure_arc.py

```diff
@@ -38,23 +39,32 @@ def __init__(self, **kwargs):
         super(AzureArcCredential, self).__init__()
 
         url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
-        if not url:
+        imds = os.environ.get(EnvironmentVariables.IMDS_ENDPOINT)
+        if not (url and imds):
             # Azure Arc managed identity isn't available in this environment
             self._client = None
             return
+
         identity_config = kwargs.pop("_identity_config", None) or {}
         config = _get_configuration()
-        client_args = dict(
-            kwargs,
+
+        self._client = ManagedIdentityClient(
             _identity_config=identity_config,
             policies=_get_policies(config),
             request_factory=functools.partial(_get_request, url),
+            **kwargs
         )
 
-        self._client = ManagedIdentityClient(**client_args)
-
     def get_token(self, *scopes, **kwargs):
         # type: (*str, **Any) -> AccessToken
+        if self._client._identity_config != {}:
+            raise ClientAuthenticationError(
+                message="User assigned identity is not supported by the Azure Arc Managed Identity Endpoint. To " \
```

Comment:
I think we should tweak the message a little. `DefaultAzureCredential` also accepts a client id for managed identity via the `managed_identity_client_id` keyword argument but I think trying to explain that in the message will mostly just make it harder to read. What do you think of "User assigned managed identities are not supported by Azure Arc. To authenticate with the system assigned identity omit the client id when constructing the credential, and if authenticating with DefaultAzureCredential ensure the AZURE_CLIENT_ID environment variable is not set."

We should also have a test to verify `get_token` raises `ClientAuthenticationError` when the credential was given `client_id`.

Reply:
I agree about the tests -- I was going to add some after looking for a reference of how to check for the error, but wanted to get eyes on other changes in the meantime. I like that comment as well; I'll make that change now